[Jayne1010] iP

Jayne1010 opened the pull request:
### Duck - Your Task Management Assistant

> "Your mind is for having ideas, not holding them." – David Allen

Duck frees your mind from the burden of remembering things you need to do. It's:

- **Text-based**
- **Easy to Learn**
- **FAST SUPER FAST to use**

All you need to do is:

1.  **Download it [[here].**](https://github.com/nus-cs2103-AY2324S2/ip/pull/96)
2.  **Double-click it.**
3.  **Add your tasks.**
4.  **Let it manage your tasks for you 😉**


And the best part? It's **FREE**!

## Features:

- [ ] **Managing Tasks**
- [ ] **Managing Deadlines**
- [ ] **Reminders (Coming Soon)**

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```Duck
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Jayne1010 committed:
Level-0

Jayne1010 committed:
Level-1

Jayne1010 committed:
Level-2

Jayne1010 committed:
Level-3

Jayne1010 committed:
Level-4

Jayne1010 committed:
Level-5

Jayne1010 committed:
Level-6

Jayne1010 committed:
test-case1

Jayne1010 committed:
Level-6

Jayne1010 committed:
Level-6

Jayne1010 committed:
Level-7

Jayne1010 committed:
Level 7

Jayne1010 committed:
Level 7

Jayne1010 committed:
Level-8

Jayne1010 committed:
Level-9

Jayne1010 committed:
A-MoreOOP

Jayne1010 committed:
A-MoreOOP

Jayne1010 committed:
A-MoreOOP

Jayne1010 committed:
A-Packages

Jayne1010 committed:
Merge branch 'add-gradle-support'
Automate project builds using Gradle

Jayne1010 committed:
A-JUnit

Jayne1010 committed:
A-JUnit

Jayne1010 committed:
A-Jar

Jayne1010 committed:
A-CodingStandard

Jayne1010 committed:
A-JavaDoc

Jayne1010 committed:
Level-9

Jayne1010 committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9
Ensure a new method and match up to coding standard

Jayne1010 committed:
A-JavaDoc

Jayne1010 committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

Jayne1010 committed:
Level-9

Jayne1010 committed:
A-CheckStyle

Jayne1010 committed:
Level-8

Jayne1010 committed:
A-CheckStyle

Jayne1010 committed:
A-CheckStyle

Jayne1010 committed:
Level-10

Jayne1010 committed:
Level-10

Jayne1010 committed:
Merge branch 'master' of https://github.com/Jayne1010/ip

Jayne1010 committed:
Level-10

yashpola commented on src/main/java/Duke/AddCommand.java:
```diff
@@ -0,0 +1,40 @@
+package Duke;
```
I think package names should be in lowercase. Similar issues elsewhere

yashpola commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+ * The Parser class is responsible for parsing user input and generating corresponding commands.
+ */
+public class Parser {
+    private static final Scanner scanner = new Scanner(System.in);
```
Since this is a constant I believe the name should be in screaming snake case i.e SCANNER. Similar issues elsewhere

yashpola commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+     * @param userInput The user input to check.
+     * @return True if the input is an exit command, false otherwise.
+     */
+    public static boolean isExitCommand(String userInput) {
```
boolean variables should be in the form of is...however this is a method so maybe it should still be a verb? Such as "checkIfExitCommand"

yashpola commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+        case "event":
+            return new AddCommand(userInput);
+        default:
+            throw new DukeException("I'm sorry, but I don't understand that command :(");
```
I would prefer an explicit //Fallthrough comment since there is no `break` statement here and I'm unsure if this is intentional or not. Similar issues elsewhere

yashpola commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,199 @@
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
```
While the imports are fairly organised, it would be even better if there were line breaks in between groups of imports such as a group for io, then nio, and so on. As well a line break in between imports and package statements

yashpola commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,199 @@
+     *
+     * @throws IOException If an error occurs while creating the file.
+     */
+    public void ensureDataFileExists() throws IOException {
```
Perhaps a better method name would be createDateFile as it informs the client that the method takes an action that alters the state of the application. createDataFileIfNotExisting is even more descriptive

yashpola commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,199 @@
+     * @param tasks The list of tasks to encode.
+     * @return The encoded list of strings.
+     */
+    public static List<String> encode(List<Task> tasks) {
```
I think the method name here can be more descriptive, to say encodeTaskToString or encodeToString so the client is clear how they are transforming the input

yashpola commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,199 @@
+            return newEvent;
+        default:
+            // Handle other task types if needed
+            return null;
```
I like how well-organised the code is here with the clear separation of logical units

yashpola reviewed (commented):
LGTM except a few nits! Great job on the first 4 sprints and the code looks readable and maintainable

Jayne1010 committed:
Level-10

Jayne1010 committed:
Level-10

Jayne1010 committed:
Merge branch 'branch-Level-10'

larainezo commented on src/main/java/Duke/DeleteCommand.java:
```diff
@@ -0,0 +1,51 @@
+ * The DeleteCommand class represents a command to delete a task from the TaskList.
+ * It implements the Command interface.
+ */
+public class DeleteCommand implements Command {
```
I think it would be better to name this CommandDeleter (according to the naming standards)

larainezo commented on src/main/java/Duke/ExitCommand.java:
```diff
@@ -0,0 +1,24 @@
+ * The ExitCommand class represents a command to exit the Duke application.
+ * It implements the Command interface.
+ */
+public class ExitCommand implements Command {
```
Suggestion to change this to CommandExiter (based on naming standards) :)

larainezo commented on src/main/java/Duke/FindCommand.java:
```diff
@@ -0,0 +1,32 @@
+ * The FindCommand class represents a command to find tasks containing a specific keyword.
+ * It implements the Command interface.
+ */
+public class FindCommand implements Command {
```
Similar to the other classes, this can be CommandFinder 👍

larainezo replied:
UnmarkCommand can be CommandUnmarker and
MarkCommand can be CommandMarker

larainezo commented on src/main/java/Duke/Command.java:
```diff
@@ -0,0 +1,19 @@
+ * on a TaskList. Classes implementing this interface must provide an execute
+ * method to carry out the specific command.
+ */
+public interface Command {
```
It's a good idea to create a Command interface! I didn't think of that 😄

larainezo reviewed (commented):
Good job! I think you code is very neat and you demonstrate good use of OOP! Keep it up :)

Jayne1010 committed:
refactor: Change package names to lowercase

Updated all package names to lowercase.

Jayne1010 committed:
refactor: Change packages names to lowercase

Updated all package names to lowercase.

Jayne1010 committed:
Add assertions for input validation

This commit adds assert statement to various command classes to validate input parameters
and ensure the corrections of assumptions made during execution.
Assertions are included to verify that crucial variables are not null.

Jayne1010 committed:
Merge pull request #2 from Jayne1010/branch-A-CodeQuality

docs: Improve code documentation with Javadoc comments

Jayne1010 committed:
docs: Improve code documentation with Javadoc comments

Added missing Javadoc comments to enhance code documentation and readability.

Jayne1010 committed:
Merge pull request #3 from Jayne1010/branch-A-CodeQuality

docs: Improve code documentation with Javadoc comments

Jayne1010 committed:
docs: Improve code documentation with Javadoc comments

Added missing Javadoc comments to enhance code documentation and readability.

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
C-Tagging

Jayne1010 committed:
docs: Update README with project overview and instructions

Jayne1010 committed:
docs: Update README with project overview and instructions

Jayne1010 committed:
Update README.md

Jayne1010 committed:
refactor: Remove dead code and unused variables

Jayne1010 committed:
Merge branch 'master' of https://github.com/Jayne1010/ip

Jayne1010 committed:
Update README.md

Jayne1010 committed:
Update README.md

Jayne1010 committed:
docs: Update header comments for clarity and consistency

Jayne1010 committed:
refactor: Refactor decodeTask method for improved readability

Jayne1010 committed:
refactor: Reduce method complexity in UI setup

Jayne1010 committed:
refactor: Reduce method complexity in UI setup

Jayne1010 committed:
Fix gradle

Jayne1010 committed:
Refactor: Enhance code style and readability

Jayne1010 committed:
Update README.md

Jayne1010 committed:
Update README.md

Jayne1010 committed:
Merge branch 'master' of https://github.com/Jayne1010/ip

Jayne1010 committed:
Fix: Resolve bugs

Jayne1010 committed:
Fix: Resolve bugs reported

Jayne1010 committed:
Fix: Resolve bugs reported
